src/main/scala/com/microsoft/hyperspace/index/serde/LogicalPlanSerDeUtils.scala

```diff
@@ -224,8 +224,9 @@ object LogicalPlanSerDeUtils {
 
   object ExtractFileFormat {
     def unapply(fileFormat: FileFormat): Option[FileFormat] = fileFormat match {
-      case CSVFileFormatWrapper => Some(new CSVFileFormat)
-      case JsonFileFormatWrapper => Some(new JsonFileFormat)
+      // Add CSVFileFormat and JsonFileFormat in below checks for backward compatibility.
+      case CSVFileFormatWrapper | _: CSVFileFormat => Some(new CSVFileFormat)
```

Comment:
remove `_`?

Reply:
or
```scala
case CSVFileFormatWrapper => Some(new CSVFileFormat)
case JsonFileFormatWrapper => Some(new JsonFileFormat)
case ParquetFileFormat | OrcFileFormat | CSVFileFormat | JsonFileFormat => Some(fileFormat)
```

Reply:
`_` can't be removed for classes. Only for case objects like CSVFileFormatWrapper. Let me know if I should go with 
```
case _: ParquetFileFormat | _: OrcFileFormat | _: CSVFileFormat | _: JsonFileFormat => Some(fileFormat)
```
But to some